src/controllers/pokemon_controller.ts

```diff
@@ -1,46 +1,59 @@
 import * as Hapi from 'hapi';
 import axios from 'axios';
-import Pokemon from '../models/Pokemon';
+import { Pokemon } from '../models';
 
 export interface ById<T> {
-  [key:number]: T
+  [key:number]: T;
 };
 
 interface PokeType {
   slot: number;
-  type: { name: string }
+  type: { name: string };
 }
 
 export default class PokemonController {
   async index(request: Hapi.Request, h: Hapi.ResponseToolkit) {
     try {
-      const res = await axios.get('https://pokeapi.co/api/v2/pokemon?limit=807');
+      let { offset = '0' , limit = '20' } = request.query;
```

Comment:
These should be numbers.  If you use hapi validation (hapi-resource-router will look at the `validate` on your controller for this), you can both validate and it'll coerce to the correct value type.
```
export default class PokemonController {
  static validate = {
    query: {
      index: Joi.object({
        offset: Joi.number().optional().default(0),
        limit: Joi.number().optional().default(20),
      }),
    },
  };
```